### PR TITLE
[frontend] Fix flag display for observables (#8635)

### LINKED ISSUE #8635: [Regression] IP location flag no longer appears
## Description

In previous versions, in the IP list view, we had the flag of the country in which the IP was located. This flag is no longer present.

Is this intentional or a bug?

## Environment

OCTI 6.3.5

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Go to "Observations > Observables"
2. Filter on "Entity type = IPv4"
3. No country flag is visible next to the red “IPv4 Address” logos (NB: check that your IPs are located in countries and that these countries have the flag set).

## Expected Output

See the flag of the country in which the IP is located as before

#### Flag missing

Screenshot 2024-10-08 095103

#### My IP

Screenshot 2024-10-08 095123

#### The flag on my Country

Screenshot 2024-10-08 095234

### PR DESCRIPTION
**Pull request by Kedae**
Closes #8635

### CODE REVIEW

File: opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx [hunk 1]
```diff
@@ -494,14 +496,49 @@ const defaultColumns: DataTableProps['dataColumns'] = {
...
+    },
   },
   operatingSystem: {
     id: 'operatingSystem',
     label: 'Operating System',
     percentWidth: 15,
     isSortable: false,
-    render: ({ operatingSystem }) => (<Tooltip title={operatingSystem?.name}><>{operatingSystem?.name ?? '-'}</></Tooltip>),
+    render: ({ operatingSystem }) => (<Tooltip title={operatingSystem?.name}><>{operatingSystem?.name ?? '-'}</>
+    </Tooltip>),
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx]:
```suggestion
    render: ({ operatingSystem }) => (
      <Tooltip title={operatingSystem?.name}>
        {operatingSystem?.name ?? '-'}
      </Tooltip>
    ),
```

File: opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx [hunk 2]
```diff
@@ -494,14 +496,49 @@ const defaultColumns: DataTableProps['dataColumns'] = {
...
+    // Please check the String.jsx->renderObservableValue. It should have the same behavior and will replace it at the end.
+    render: (observable, helpers) => {
+      switch (observable.entity_type) {
+        case 'IPv4-Addr':
+        case 'IPv6-Addr': {
+          const country = observable.countries?.edges?.at(0)?.node;
+          if (country) {
+            const flag = R.head(
+              (country.x_opencti_aliases ?? []).filter((n: string) => n.length === 2),
+            );
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx]:
```suggestion
            const flag = (country.x_opencti_aliases ?? []).filter((n: string) => n.length === 2)[0];
```
Ramda head is not exactly rocket science :)

File: opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx [hunk 3]
```diff
@@ -494,14 +496,49 @@ const defaultColumns: DataTableProps['dataColumns'] = {
...
+            if (flag) {
+              return (
+                <div>
+                  <div style={{ float: 'left', paddingTop: 2 }}>
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx]: better use flexbox ?

Comment (Kedae) [reply opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx]: Done

File: opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx (hunk 2 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx]: Done

File: opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx (hunk 1 above)
Comment (Kedae) [reply opencti-platform/opencti-front/src/components/dataGrid/dataTableUtils.tsx]: Done

**Review by labo-flg — APPROVED** [review]

**Pull request merged**